test/integration/go.mod

```diff
@@ -8,8 +8,9 @@ require (
 	k8s.io/apimachinery v0.17.17
 	k8s.io/client-go v0.17.17
 	k8s.io/klog/v2 v2.8.0
-	k8s.io/utils v0.0.0-20210305010621-2afb4311ab10 // indirect
+	k8s.io/utils v0.0.0-20210305010621-2afb4311ab10
 	metacontroller.io v0.0.0-00010101000000-000000000000
+	sigs.k8s.io/controller-runtime v0.5.4
```

Comment:
The reason for using v0.5.4 is because it requires the same kubernetes library versions as the ones defined in metacontroller's `go.mod`. As soon as we update our own kubernetes library versions, we can upgrade controller-runtime as well

Reply:
oh, this raises a important question, does controller-runtume limits somehow on which k8s versions it can run ? As currently metacontrollers support vast range of k8s versions, by not depending on k8s API directly.

Reply:
This is a good question. I am not aware of any limitation controller-runtime itself has outside of the ones that come from the kubernetes libraries. Seeing green e2e tests is also a good indication that things would continue working with older versions. 

> As currently metacontrollers support vast range of k8s versions, by not depending on k8s API directly.

Could you elaborate a bit on this point? How does metacontroller avoid a direct dependency on the k8s api?

Reply:
currently we depend on k8s core libraries but nothing related to typing, except `Events`, but their api was stable from early version of k8s

Reply:
all of the code usues `Unstructured` when recieving and sending types to API server

Reply:
but I think that for metacontroller resources we are safe - those CRD's are defined in the controller itself. 

As for the rest - maybe there is no point keeping resources in older version - I decoupled test dependencies to test sub-project, I will test if we are really independent